phoenix-core/src/it/java/org/apache/phoenix/end2end/WhereOptimizerForArrayAnyIT.java

```diff
@@ -0,0 +1,807 @@
+        try (ResultSet rs = stmt.executeQuery(selectSql)) {
+          assertTrue(rs.next());
+          assertEquals(DateUtil.parseTimestamp(pk1Value), rs.getTimestamp(1));
+          assertEquals(DateUtil.parseDate(pk2Value), rs.getDate(2));
+          assertEquals("a", rs.getString(3));
+        }
+      }
+      assertPointLookupsAreGenerated(selectSql, conn, null);
+    }
+  }
+
+  @Test
+  public void testArrayAnyComparisonForBinaryColumn() throws Exception {
+    String tableName = generateUniqueName();
+    String ddl = "CREATE TABLE " + tableName + " (" + "pk1 BINARY(3) NOT NULL, "
+      + "pk2 VARBINARY_ENCODED(3) NOT NULL, " + "pk3 VARBINARY(3) NOT NULL, " + "col1 VARCHAR, "
+      + "CONSTRAINT pk PRIMARY KEY (pk1, pk2, pk3)" + ")";
+    try (Connection conn = DriverManager.getConnection(getUrl())) {
+      try (Statement stmt = conn.createStatement()) {
+        stmt.execute(ddl);
+        conn.commit();
+      }
+    }
+    String pk1Value = "a";
+    String pk2Value = "b";
+    String pk3Value = "c";
+    String col1Value = "d";
+    String upsertStmt = "UPSERT INTO " + tableName + " VALUES (?, ?, ?, ?)";
+    try (Connection conn = DriverManager.getConnection(getUrl())) {
+      try (PreparedStatement stmt = conn.prepareStatement(upsertStmt)) {
+        stmt.setBytes(1, pk1Value.getBytes());
+        stmt.setBytes(2, pk2Value.getBytes());
+        stmt.setBytes(3, pk3Value.getBytes());
+        stmt.setString(4, col1Value);
+        stmt.executeUpdate();
+        conn.commit();
+      }
+    }
+    try (Connection conn = DriverManager.getConnection(getUrl())) {
+      TestUtil.dumpTable(conn, TableName.valueOf(tableName));
+      String selectSql =
+        "SELECT * FROM " + tableName + " WHERE pk1 = ANY(?) AND pk2 = ANY(?) AND pk3 = ANY(?)";
+      byte[][] nativeByteArr =
+        new byte[][] { pk1Value.getBytes(), pk2Value.getBytes(), pk3Value.getBytes() };
+      Array binaryArr = conn.createArrayOf("BINARY", nativeByteArr);
+      Array varbinaryArr = conn.createArrayOf("VARBINARY", nativeByteArr);
+      try (PreparedStatement stmt = conn.prepareStatement(selectSql)) {
+        stmt.setArray(1, binaryArr);
+        stmt.setArray(2, varbinaryArr);
+        stmt.setArray(3, varbinaryArr);
+        try (ResultSet rs = stmt.executeQuery()) {
+          assertTrue(rs.next());
+          assertBinaryValue(pk1Value.getBytes(), rs.getBytes(1));
+          assertBinaryValue(pk2Value.getBytes(), rs.getBytes(2));
+          assertBinaryValue(pk3Value.getBytes(), rs.getBytes(3));
+          assertEquals(col1Value, rs.getString(4));
+        }
+      }
+      try (PreparedStatement stmt = conn.prepareStatement("EXPLAIN " + selectSql)) {
+        stmt.setArray(1, binaryArr);
+        stmt.setArray(2, varbinaryArr);
+        stmt.setArray(3, varbinaryArr);
+        try (ResultSet rs = stmt.executeQuery()) {
+          String explainPlan = QueryUtil.getExplainPlan(rs);
+          assertTrue(explainPlan.contains("POINT LOOKUP ON"));
```

Comment:
For each of these places, since we just need to scan type, let's use `ExplainPlanAttributes` based assertion:
```
    ExplainPlan plan = stmt.unwrap(PhoenixPreparedStatement.class).optimizeQuery().getExplainPlan();
    ExplainPlanAttributes planAttributes = plan.getPlanStepsAsAttributes();
    assertEquals("POINT LOOKUP ON N KEY ", planAttributes.getExplainScanType());
```

Reply:
Sure. One doubt: Can you please tell why is this better than existing approach? Thanks

Reply:
Got it, thanks.